### PR TITLE
Add documentation for [DeploymentItem] attribute

### PR DESCRIPTION
**Pull request by engyebrahim**
fix: https://github.com/microsoft/testfx/issues/1320 and https://github.com/microsoft/testfx/issues/676

### CODE REVIEW

**Comment by engyebrahim** [comment]
Hi @Mikejo5000, please wait before merging until @Evangelink approve too.

**Comment by Court72** [comment]
@Mikejo5000, @Evangelink  

Can you review the proposed changes? 

When the changes are ready for publication, add a `#sign-off` comment to signal that the PR is ready for the review team to merge. 

#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

File: docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md
Comment (Mikejo5000) [inline docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md:362]:
```suggestion
And here you can see its [constructors](<xref:Microsoft.VisualStudio.TestTools.UnitTesting.DeploymentItemAttribute#constructors>).
```

Comment (Mikejo5000) [reply docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md:362]: Looks like the xref is case sensitive. Also, I don't recall seeing bookmark/anchor links in an xref, so you might have to remove the "#constructors" portion to get the link to work correctly.

File: docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md
Comment (Evangelink) [inline docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md:356]: What is starting with MSTest v2? Is it the attribute or the copy or all?

Comment (Evangelink) [inline docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md:356]:
```suggestion
The <xref:Microsoft.VisualStudio.TestTools.UnitTesting.DeploymentItemAttribute> for copying files or folders specified as deployment items to the deployment directory. Deployment directory is where all the deployment items are present along with test project dll. (starting from MSTest v2)
```

File: docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md
Comment (Evangelink) [inline docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md:367]: I would remove this comment here and add it in the explanation above (see previous comment).

File: docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md
Comment (Evangelink) [inline docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md:369]:
```suggestion
[DeploymentItem(@"C:\classLevelDepItem.xml")]   // Copy file using some absolute path
```

File: docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md
Comment (Evangelink) [inline docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md:373]:
```suggestion
    [DeploymentItem(@"..\..\methodLevelDepItem1.xml")]   // Copy file using a relative path from the dll output location
```

File: docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md
Comment (Evangelink) [inline docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md:374]:
```suggestion
    [DeploymentItem(@"C:\DataFiles\methodLevelDepItem2.xml", "SampleDataFiles")]   // File will be added under a SampleDataFiles in the deployment directory
```

**Review by Evangelink — APPROVED** [review]

**Review by Evangelink — APPROVED** [review]

File: docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md
Comment (Mikejo5000) [inline docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md:356]:
```suggestion
The MSTest V2 framework introduced <xref:Microsoft.VisualStudio.TestTools.UnitTesting.DeploymentItemAttribute> for copying files or folders specified as deployment items to the deployment directory (without adding a custom output path the copied files will be in TestResults folder inside the project folder). The deployment directory is where all the deployment items are present along with test project DLL.
```

**Review by Mikejo5000 — CHANGES_REQUESTED** [review]
@engyebrahim Minor edits suggested. Is this PR also waiting for the MSTest v4 update? Or is it ready to merge?

**Comment by engyebrahim** [comment]
Hi @Mikejo5000, this one is ready to merge

**Review by Mikejo5000 — APPROVED** [review]

**Comment by Mikejo5000** [comment]
#sign-off Thanks!

**Pull request merged**
